wfe/wfe.go

```diff
@@ -983,8 +984,11 @@ func (wfe *WebFrontEndImpl) Challenge(
 
 	authz, err := wfe.SA.GetAuthorization(ctx, authorizationID)
 	if err != nil {
-		// TODO(#1198): handle db errors etc
-		notFound()
+		if err == sql.ErrNoRows {
+			notFound()
+		} else {
+			wfe.sendError(response, logEvent, probs.ServerInternal("Internal Server Error"), err)
```

Comment:
`unable to find authorization` is too specific and more related to a status 404 instead of a 500. The `else` case  it's an unexpected error (like database connection or something like that) so we should return a 500.